apis/python/src/tiledbsoma/_experiment.py

```diff
@@ -74,3 +73,26 @@ def _set_create_metadata(cls, handle: Wrapper[Any]) -> None:
         # TileDB Cloud UI to detect that they are SOMA datasets.
         handle.metadata["dataset_type"] = "soma"
         return super()._set_create_metadata(handle)
+
+    # in tiledbsoma
+    def axis_query(  # type: ignore
+        self,
+        measurement_name: str,
+        *,
+        obs_query: Optional[query.AxisQuery] = None,
+        var_query: Optional[query.AxisQuery] = None,
+    ):
+        """Creates an axis query over this experiment.
+
+
+        Lifecycle: maturing
+        """
+        # mypy doesn't quite understand descriptors so it issues a spurious
+        # error here.
+        return query.ExperimentAxisQuery(  # type: ignore
+            self,
+            measurement_name,
+            obs_query=obs_query or query.AxisQuery(),
+            var_query=var_query or query.AxisQuery(),
+            index_factory=build_index,  # comment this line to disable thw C++ indexer
```

Comment:
Hi @thetorpedodog I managed to implemented all of your requested changes except for this one because it continuously fails this after that change:
```
$ python  libtiledbsoma/src/reindexer/example.py
99187
Traceback (most recent call last):
  File "/Users/brobatmili/projects/TileDB-SOMA/libtiledbsoma/src/reindexer/example.py", line 26, in <module>
    main()
  File "/Users/brobatmili/projects/TileDB-SOMA/libtiledbsoma/src/reindexer/example.py", line 21, in main
    query.to_anndata(X_name="raw")
  File "/Users/brobatmili/soma_index_abstraction/lib/python3.11/site-packages/somacore/query/query.py", line 294, in to_anndata
    return self._read(
           ^^^^^^^^^^^
  File "/Users/brobatmili/soma_index_abstraction/lib/python3.11/site-packages/somacore/query/query.py", line 401, in _read
    x_matrices = {
                 ^
  File "/Users/brobatmili/soma_index_abstraction/lib/python3.11/site-packages/somacore/query/query.py", line 402, in <dictcomp>
    _xname: _fast_csr.read_csr(
            ^^^^^^^^^^^^^^^^^^^
  File "/Users/brobatmili/soma_index_abstraction/lib/python3.11/site-packages/somacore/query/_fast_csr.py", line 29, in read_csr
    acc = _CSRAccumulator(
          ^^^^^^^^^^^^^^^^
  File "/Users/brobatmili/soma_index_abstraction/lib/python3.11/site-packages/somacore/query/_fast_csr.py", line 98, in __init__
    self.obs_indexer = index_factory(self.obs_joinids, context)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: build_index() got multiple values for argument 'context'
```
Appreciate any suggestions.

Reply:
In this case you will want to go back to `git+https://github.com/single-cell-data/SOMA.git@1d71cc1f54eb700f6215d1d82cfa74e0d01571d3` as the dependency version for somacore.

Reply:
It works. Thanks